core/modules/compare/components/Product.ts

```diff
@@ -5,8 +5,8 @@ export const CompareProduct = {
   name: 'CompareProduct',
   mixins: [compareMountedMixin],
   computed: {
-    isOnCompare (): boolean {
-      return this.$store.getters['compare/isOnCompare'](this.product)
+    isToCompare (): boolean {
+      return this.$store.getters['compare/isToCompare'](this.product)
```

Comment:
you've changed name here but did not changed actual getter in core/modules/compare/store/getters.ts

Reply:
More proper name would be isCompared or isBeingCompared. isToCompare sounds weird. isOnCompare is also valid because it's a shortened version of isOnCompareList. Why did you decide to change the naming in the first place?

Anyway, if you decide to change the naming then it's a BC and you have to include information about that in upgrade notes that are in docs.